src/main/kotlin/uk/gov/justice/digital/hmpps/manageoffencesapi/service/AdminService.kt

```diff
@@ -75,6 +83,60 @@ class AdminService(
     }
   }
 
+  @Transactional
+  fun createEncouragementOffence(parentOffenceId: Long): Offence {
+    val offence = offenceRepository.findById(parentOffenceId)
+      .orElseThrow { EntityNotFoundException("Offence not found with ID $parentOffenceId") }
+
+    if (offence.parentCode !== null && offence.parentCode == offence.code) {
+      throw ValidationException("Offence must be a valid parent")
+    }
+
+    if (offence.endDate !== null && offence.endDate < encouragementOffenceEligibilityStartDate) {
```

Comment:
I'm not 100% this logic is correct. But this seemed the safest approach.

Reply:
looks right to me